internal/bminventory/inventory.go

```diff
@@ -525,7 +515,15 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 		TriggerMonitorTimestamp:     time.Now(),
 		MachineNetworkCidrUpdatedAt: time.Now(),
 	}
+	if params.NewClusterParams.OlmOperators != nil {
+		var newOLMOperators []*models.MonitoredOperator
+		newOLMOperators, err = b.getOLMOperators(&cluster, params.NewClusterParams.OlmOperators)
+		if err != nil {
+			return nil, err
+		}
 
+		monitoredOperators = append(monitoredOperators, newOLMOperators...)
```

Comment:
This code won't have any effect now that you moved it down.  cluster.MonitoredOperators isn't getting the updated value of monitoredOperators.

Reply:
Now, I'm appending it to cluster.MonitoredOpeators we should be good now.